substrate/client/rpc/src/state/state_full.rs

```diff
@@ -166,6 +166,26 @@ where
 		}
 		Ok(())
 	}
+
+	/// Helper function to check a key against lists of prefixes
+	fn is_target_key(
+		&self,
+		key_to_check: StorageKey,
+		include_prefixes : Option<Vec<StorageKey>>,
+	) -> bool {
+		if let Some(prefixes_to_include) = include_prefixes {
+			for include_prefix in prefixes_to_include {
+				if key_to_check.0.starts_with(&include_prefix.0) {
+					return true
+				}
+			}
+			// skip all keys that do not have any of "included prefixes"
+			return false
+		}
+
+		// by default process all keys
+		return true
+	}
```

Comment:
let's please put back the param `excluded_prefixes` because it will allow to reduce the amount of data read from the node by the indexer significantly.